### DISCUSSION TITLE
How to invalidate a cloudfront path that contains a tilde ~ character

### DISCUSSION THREAD

**Discussion by danludwig-happymoney** [aws/aws-sdk-js#4228] [Q&A]
If you create a normal cloudfront distribution with an s3 origin, and put a file in the bucket named "some~thing.txt", that file can be accessed from the cloudfront.net domain as its typed in this message. The tilde character does not need to be URL encoded. 

However, if you try to create a cloudfront invalidation with the path `/some~thing.txt`, cloudfront complains that 
> Your request contains one or more invalid invalidation paths.

Is it not possible to invalidate a cloudfront path that contains a tilde character? Or is there some magic undocumented incantation to make it work?

**Comment by ajredniwja** [comment]
Hey @danludwig-happymoney thanks for opening this issue, I believe I have seen this issue before when special characters being used in creating the invalidation. I have reached out to the cloudfront team to know more when they can work on it thanks.
